Added ability to rotate player positions in the local game view.

mildmongrel opened the pull request:
Added menu items (and keyboard shortcuts) to the game menu to rotate the local game view.  This is the proposed replacement for PR #1175.

I added this for multiplayer games - in the two-column view the local player always ended up in the bottom left corner. In 2v2 multiplayer this setting allows players to adjust their "seat at the table".

ZeldaZach commented:
I tested this and there seems to be a small issue with rotations.

When you rotate and the boards aren't the same size, resizing comes into play:
![screenshot 2015-06-29 18 41 05](https://cloud.githubusercontent.com/assets/7460172/8420250/6dbe674c-1e8e-11e5-9433-93e9b57e6dfa.png)
![screenshot 2015-06-29 18 40 58](https://cloud.githubusercontent.com/assets/7460172/8420249/6dbdd502-1e8e-11e5-8cc3-979b422f9b9c.png)

mildmongrel commented:
OK, had a chance to look at this further and I think the layout thing is a separate issue that should be resolved with a different PR.

Also I found a bug I introduced on this branch so let's hold off on merging this until I've fixed it and the layout issue is resolved.

woogerboy21 commented:
Do does #1189 also address the "bug" you mentioned or is that still being worked on?

mildmongrel commented:
No, #1189 fixes the layout only.  The other bug is layout-related though - if the layout fix is accepted I want to rebase and retest this PR to make sure everything's hunky dory.

Fizztastic commented:
Tested in local game and online, works as expected.

ZeldaZach commented:
Meets my expectations for sure :shipit:

Daenyth commented:
Can you squash the commits into one?

Daenyth commented on cockatrice/src/gamescene.cpp:
```diff
@@ -44,39 +48,63 @@ void GameScene::addPlayer(Player *player)
+        Player *p = playersIter.next();
+        if (!p->getConceded()) {
+            playersPlaying.append(p);
+            if (!firstPlayerFound && (p->getLocal())) {
```
Is it not true that there will be either 0 or 1 players where `getLocal() == true`? We might not need the found flag.

mildmongrel replied:
Previous logic had the same kind of check and I thought the same when I started working on this.  Turns out when you start a local game all players are considered local.

Daenyth replied:
I wish we could remove local mode, but too many people use it. Good catch.

Daenyth commented:
Can we test with a large number of players, hit a bunch of rotations, have players concede, rotate more, etc, and make sure it all continues to work as expected?

ZeldaZach commented:
I'll check it out later tonight and use the additional testing options suggested.

ZeldaZach commented:
Works fully expected. I tested in an 8 player game and had people concede while I was rotating around left & right. Great job @mildmongrel!! :shipit:

Daenyth commented:
This is an awesome feature!

On Wed, Jul 8, 2015 at 6:23 PM Zach <notifications@github.com> wrote:

> Merged #1179 <https://github.com/Cockatrice/Cockatrice/pull/1179>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Cockatrice/Cockatrice/pull/1179#event-351298685>.
>
